src/Streamly/Coreutils/String.hs

```diff
@@ -0,0 +1,24 @@
+-- |
+-- Module      : Streamly.Coreutils.String
+-- Copyright   : (c) 2022 Composewell Technologies
+-- License     : BSD-3-Clause
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+module Streamly.Coreutils.String
+    (
+        deIndent
+    )
+where
+
+import Data.Char (isSpace)
+
+-- | Removes the common minimum space at beginning of each line in a string
+--
+deIndent :: String -> String
+deIndent input =
+    let lns = lines input
+        indent = minimum $ fmap (length . takeWhile isSpace) lns
+        output = unlines $ fmap (drop indent) lns
+     in output
```

Comment:
We should ignore the blank lines. If there are blank lines in the text the common minimum will become zero which is not what we want.